source/PlayerEntity.cpp

```diff
@@ -147,6 +155,58 @@ void PlayerEntity::onSimulation(SimTime absoluteTime, SimTime deltaTime) {
 			respawn();
 		}
 	}
+
+	if (!m_gettingMovementInput) {
+
+		if (accelerationEnabled!= nullptr && *accelerationEnabled) {
+			m_acceleratedVelocity = lerp(m_acceleratedVelocity, 0.0f, *movementDeceleration * deltaTime);
+		}
+		else {
+			m_acceleratedVelocity = 0;
+		}
+
+		if (m_acceleratedVelocity > 0) {
+			m_linearVector = m_acceleratedVelocity * m_lastDirection;
+		}
+
+		if (headBobEnabled!= nullptr && *headBobEnabled) {
+			m_headBobCurrentHeight = lerp(m_headBobCurrentHeight, 0.0f, 0.1f);
+		}
+	}
+	else {
+
+		if (*accelerationEnabled) {
+			m_acceleratedVelocity = lerp(m_acceleratedVelocity, m_walkSpeed, *movementAcceleration * deltaTime);
```

Comment:
I don't believe the units on this math work out correct for `movementAcceleration`/`movementDeceleration` being provided in m/s^2. `lerp(a,b,c)` does `a + (b-a)*c` so if `c` was provided in m/s^2 * s = m/s then the `(b-a)*c` term would have units m/s * m/s = (m/s)^2 which is not correct.

Currently I believe this results in a sort of logarithmic acceleration profile. If `deltaTime*movementAcceleration/Deceleration` is always <1 this becomes asymptotic to the specified move speed (the speed will never actually reach the rate). 

If it were me I'd move away from using `lerp()` here and instead use:
`m_acceleratedVelocity = min(m_acceleratedVelocity + movementAcceleration * deltaTime, m_walkSpeed)`
and
`m_acceleratedVelocity = max(m_acceleratedVelocity - movementDeceleration * deltaTime, m_walkSpeed)`
this should guarantee linear acceleration up to the min/max velocity.

Explainability of exact mechanics of systems in FPSci has become a top priority for us. We started w/ a more "looks right" model but have been bitten by it several times when it comes to explaining exactly how a mechanic worked for reproducibility in a paper.

Reply:
Thank you for the sugeestions. I have made the necessary changes.